src/main/java/io/zold/api/RtTransaction.java

```diff
@@ -77,27 +84,51 @@
     /**
      * String representation of transaction.
      */
-    private final String transaction;
+    private final Scalar<String> transaction;
 
     /**
      * Ctor.
      * @param trnsct String representation of transaction
-     * @todo #15:30min Check validity of the incoming string representation
-     *  of transaction. It should comply the pattern described in the white
-     *  paper. Cover with unit tests.
      */
     RtTransaction(final String trnsct) {
-        this.transaction = trnsct;
+        this.transaction = new StickyScalar<String>(
+            () -> {
+                if (
+                    new TrimmedText(new TextOf(trnsct)).asString().length() < 1
+                ) {
+                    throw new IOException(
+                        "Invalid transaction string: string is empty"
+                    );
+                }
+                final List<Text> pieces =
+                    new StickyList<>(
+                        new ListOf<>(
+                            new SplitText(trnsct, ";")
+                        )
+                    );
+                // @checkstyle MagicNumberCheck (1 line)
+                if (pieces.size() != 7) {
+                    throw new IOException(
+                        new FormattedText(
+                            // @checkstyle LineLength (1 line)
+                            "Invalid transaction string: expected 7 fields but have %d",
```

Comment:
@paulodamaso minor comment regarding the English. Let's change it to: `Invalid transaction string, expected 7 fields, but found %d`